### PR TITLE
chore: make Header encoding good

### PR DESCRIPTION
**Pull request by Rjected**
## Motivation

Previous code had a bunch of incorrect placeholders only required for arbitrary impls that output incorrect headers.

## Solution

Remove the placeholders

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/consensus/src/header.rs [hunk 1]
```diff
@@ -272,53 +272,32 @@ impl Header {
         length += self.timestamp.length();
         length += self.extra_data.length();
         length += self.mix_hash.length();
-        length += self.nonce.length();
+        length += B64::new(self.nonce.to_be_bytes()).length();
```
Comment (mattsse) [inline crates/consensus/src/header.rs:275]: this seems redundant?

Comment (Rjected) [reply crates/consensus/src/header.rs]: artifact of reth impl using u64 nonce

File: crates/consensus/src/header.rs [hunk 2]
```diff
@@ -390,63 +369,31 @@ impl Encodable for Header {
         self.timestamp.encode(out);
         self.extra_data.encode(out);
         self.mix_hash.encode(out);
-        self.nonce.encode(out);
+        B64::new(self.nonce.to_be_bytes()).encode(out);
```
[the remark this answers is not in the document]
Comment (Rjected) [reply crates/consensus/src/header.rs]: artifact from copypasta

File: crates/consensus/src/header.rs [hunk 3]
```diff
@@ -488,55 +433,30 @@ impl Decodable for Header {
...
-        // Decode requests hash.
-        //
-        // If new fields are added, the above pattern will need to
-        // be repeated and placeholders decoded. Otherwise, it's impossible to tell _which_
-        // fields are missing. This is mainly relevant for contrived cases where a header is
-        // created at random, for example:
-        //  * A header is created with a withdrawals root, but no base fee. Shanghai blocks are
-        //    post-London, so this is technically not valid. However, a tool like proptest would
-        //    generate a block like this.
+        // Decode requests root.
```
Comment (mattsse) [inline crates/consensus/src/header.rs:459]:
```suggestion
        // Decode requests hash.
```

**Review by mattsse — APPROVED** [review]

**Pull request merged**